subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/artifact/ArtifactBackedResolvedVariant.java

```diff
@@ -80,6 +84,11 @@ public AttributeContainerInternal getAttributes() {
         return attributes;
     }
 
+    private static boolean isFromIncludedBuild(ResolvedArtifact artifact) {
+        ComponentArtifactIdentifier id = artifact.getId();
+        return id instanceof LocalComponentArtifactMetadata && ((LocalComponentArtifactMetadata) id).isComposite();
+    }
+
```

Comment:
This should work, and avoids the need to add the `LocalComponentArtifactMetadata.isComposite()` method:

```
    private static boolean isFromIncludedBuild(ResolvedArtifact artifact) {
        ComponentIdentifier id = artifact.getId().getComponentIdentifier();
        return id instanceof ProjectComponentIdentifier 
            && !((ProjectComponentIdentifier) id).getBuild().isCurrentBuild();
    }
```